R/utils.R

```diff
@@ -288,28 +288,47 @@ check_attributes <- function(attributes, call = caller_env()) {
 }
 
 # apple + spark drive config (#651) --------------------------------------------
-configure_spark <- function(call = caller_env()) {
+# Method will attempt to:
+# 1. Locate the simba driver config using the user supplied callback.  Callback
+# method defaults to locate_config_spark.
+# 2. Inspect the config for some settings that can impact how our package
+# performs.
+# 3. If action == "modify" then we attempt to modify the config in-situ.
+# 4. Otherwise we throw a warning asking the user to revise.
+configure_simba <- function(locate_config_callback = (function() character()),
+                            action = "modify", call = caller_env()) {
   if (!is_macos()) {
     return(invisible())
   }
-
+  if (is.null(getOption("odbc.no_config_override"))) {
+    return(invisible())
+  }
   unixodbc_install <- locate_install_unixodbc()
   if (length(unixodbc_install) == 0) {
     error_install_unixodbc(call)
   }
 
-  spark_config <- locate_config_spark()
-  if (length(spark_config) == 0) {
-    abort(
+  simba_config <- locate_config_callback()
+  if (length(simba_config) == 0) {
+    func <- cli::warn
+    if ( action == "modify" ){
```

Comment:
```suggestion
    if (action == "modify"){
```

[Tidy style](https://style.tidyverse.org/syntax.html)--these changes apply to a few places in this PR, incl for before and after curly braces as well.

Reply:
Thanks! Still looking for a good tidy style vim plugin that works in my existing configuraion.  Done.